portfolio/src/main/webapp/script.js

```diff
@@ -35,3 +35,22 @@ function getHello() {
     document.getElementById('hello-container').innerText = quote;
   });
 }
+
+function getMessages() {
+  fetch('/data').then(response => response.json()).then((messages) => {
+    const messageListElement = document.getElementById("messages-container");
+    messageListElement.innerHTML = '';
+    var i;
+    for (i = 0; i < messages.length; i++) {
```

Comment:
I'd suggest using `let` instead of `var`. The main difference is that `let` is block-scoped whereas `var` is function-scoped: https://stackoverflow.com/questions/762011

Additionally, you can declare the variable inside the for loop so:
`for (let i = 0; i < messages.length; i++)`

Reply:
didn't know that! will do